sdk/src/bigint.rs

```diff
@@ -1,17 +1,43 @@
 use core::{
     cmp::Ordering,
+    fmt::Debug,
     ops::{Add, BitAnd, BitOr, BitXor, Div, Mul, Neg, Not, Rem, Shl, Shr, Sub},
 };
 
 use super::{
-    env::internal::Env as _, xdr::ScObjectType, ConversionError, Env, EnvBase, EnvObj, EnvVal,
-    RawVal, TryFromVal,
+    env::internal::Env as _, xdr::ScObjectType, Binary, ConversionError, Env, EnvBase, EnvObj,
+    EnvVal, RawVal, RawValConvertible, TryFromVal,
 };
 
 #[repr(transparent)]
 #[derive(Clone)]
 pub struct BigInt(EnvObj);
 
+impl Debug for BigInt {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        let env = self.env();
+        let bi = self.0.to_tagged();
+        let obj: EnvObj = env.bigint_to_radix_be(bi, 10u32.into()).in_env(env);
+        write!(f, "BigInt(")?;
+        if let Ok(bin) = TryInto::<Binary>::try_into(obj) {
+            let sign = env.bigint_cmp(bi, env.bigint_from_u64(0));
+            match unsafe { <i32 as RawValConvertible>::unchecked_from_val(sign) } {
+                -1 => write!(f, "-")?,
+                0 => {
+                    write!(f, "0)")?;
+                    return Ok(());
```

Comment:
This feels like a gotcha that we exit early here and append the `)` and not on other cases. Is it necessary to exit early here? Looks like it should work fine to continue since the binary should be empty?

Reply:
You are right. Fixed.